### PR TITLE
refactor: use buffer from and alloc instead of buffer constructor (updated)

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

File: src/ntlm-payload.js [hunk 1]
```diff
@@ -150,20 +150,20 @@ class NTLMResponsePayload {
...
     const hash = this.ntHash(password);
-    const identity = new Buffer(user.toUpperCase() + domain.toUpperCase(), 'ucs2');
+    const identity = Buffer.from(user.toUpperCase() + domain.toUpperCase(), 'ucs2');
     return this.hmacMD5(identity, hash);
   }
 
   ntHash(text: string) {
-    const hash = new Buffer(21).fill(0);
+    const hash = Buffer.alloc(21).fill(0);
     hash.fill(0);
```
Comment (kibertoad) [inline src/ntlm-payload.js]: Is there any good reason to do fill twice, or this should be safe to be rewritten as `Buffer.alloc(21, 0)` with `hash.fill(0);` dropped?

Comment (kibertoad) [reply src/ntlm-payload.js]: OK, pushed the change

**Comment by kibertoad** [comment]
@arthurschreiber All systems green! Is there anything left?

**Comment by arthurschreiber** [comment]
@Hadis-Fard @kibertoad  Thanks to both of you for working on this! 🙇🏻‍♂️

**Pull request merged**